components/engine_traits/src/misc.rs

```diff
@@ -37,59 +47,135 @@ pub trait MiscExt: Iterable + WriteBatchExt + CFNamesExt {
 
     fn delete_all_in_range(
         &self,
+        strategy: DeleteStrategy,
         start_key: &[u8],
         end_key: &[u8],
-        use_delete_range: bool,
     ) -> Result<()> {
         if start_key >= end_key {
             return Ok(());
         }
 
         for cf in self.cf_names() {
-            self.delete_all_in_range_cf(cf, start_key, end_key, use_delete_range)?;
+            self.delete_all_in_range_cf(cf, strategy.clone(), start_key, end_key)?;
         }
 
         Ok(())
     }
 
+    fn delete_all_in_range_cf_by_ingest(
```

Comment:
let's move the logic of `delete_all_in_range_cf` into `engine_rocks` and remove `delete_all_in_range_cf_by_ingest` form `MiscExt

Reply:
Ok.

Reply:
No. delete_all_in_range_cf_by_ingest will be used in `worker/region.rs`.